### PR TITLE
feat(cmf): customize sentry initialization

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
We can't customize the sentry init and we can't call configureScope in sentry.

**What is the chosen solution to this problem?**

Customize the Sentry initialization for the hosted APP.
add a function configureSentryScope to customize the data send from the APP to Sentry. By this way, set this function on UI, we use the same Sentry instance.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/src/onError.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ function hasReportFeature() {
  * @param {Error} error instance of Error
  */
 function report(error, options = {}) {
-	if (ref.SENTRY_DSN) {
+	if (ref.SENTRY_DSN || getCurrentHub().getClient() !== undefined) {
```
Comment (jmfrancois) [inline packages/cmf/src/onError.js]: there are many places where we test ref.SENTRY_DSN to consider the sentry setup.
Not sure what is the best but here the fix is imcomplete. 
For example hasReportFeature will report false but sentry is available... so the Error Component will display wrong info.

File: packages/cmf/src/onError.js [hunk 2]
```diff
@@ -139,13 +139,13 @@ function onJSError(event) {
  * init Sentry lib
  * @return {[type]} [description]
  */
-function setupSentry() {
+function setupSentry(option = {}) {
```
Comment (sgendre) [inline packages/cmf/src/onError.js]:
```suggestion
function setupSentry(options = {}) {
```

File: packages/cmf/src/onError.js [hunk 3]
```diff
@@ -139,13 +139,13 @@ function onJSError(event) {
...
 	if (!ref.SENTRY_DSN) {
 		return;
 	}
 	window.removeEventListener('error', onJSError);
 	try {
-		init({ dsn: ref.SENTRY_DSN });
+		init({ dsn: ref.SENTRY_DSN, ...option });
```
Comment (sgendre) [inline packages/cmf/src/onError.js]:
```suggestion
		init({ dsn: ref.SENTRY_DSN, ...options });
```

File: packages/cmf/src/onError.md [hunk 4]
```diff
@@ -65,15 +65,15 @@ in your own components in some key places.
 import { ErrorBoundary } from '@talend/react-cmf';
 
 function DoComplexComponent(props) {
-    // ... do some stuff
+	// ... do some stuff
```
Comment (sgendre) [inline packages/cmf/src/onError.md]: 4 spaces vs 8 spaces ?

Comment (jmfrancois) [reply packages/cmf/src/onError.md]: not it was tab and now it space, don t know why ... may be a prettier which do not understand this is javascript :trollface:

**Pull request merged**